breadcrumbs-ui.js

```diff
@@ -0,0 +1,51 @@
+define([
+    '$'
+], function($) {
+
+    var init = function() {
+        var $breadcrumbs = $('.c-breadcrumbs');
+
+        $breadcrumbs.each(function() {
```

Comment:
The way Adaptive works is it will call this init function on each instance of the component, so you don’t need to interate over all instances here. However, I think this highlights a gap in our docs that we should fix.

I would prefer to follow the pattern set out in the Select component’s UI script, where we have a constructor for the component, and the init method either returns the existing instance (stored as jQuery data on the root element) or creates a new instance, storing it in jQuery data. That would look like this, for Breadcrumbs:

```javascript
/**
 * Constructor
 */
var Breadcrumbs = function($el, options) {
    // Component instance properties.
    this.$el = $el;
    this.$items = this.$el.find('.c-breadcrumbs__item');
    // Add event handlers here.
    // Set up initial state here.
}

/**
 * Export the init method required by AdaptiveJS (this is boilerplate).
 */
return {
    'init': function($el, options) {
        // If already initialized, return the instance; otherwise, create it
        // and expose it through `$('.c-breadcrumbs').data('component')`.
        return $el.data('component') || $el.data('component', new Breadcrumbs($el, options));
    }
};
```